apps/core/scenes/results/ResultsListItem.js

```diff
@@ -1,61 +1,246 @@
+
+// @TODO get from config
+const dateFormat = 'ccc d LLL';
+
+class ResultListItem extends React.Component<Props> {
+  // @TODO get from utils
+  getDateTimeInterval = (
+    to: DateTime,
+    from: DateTime,
+    unit: string | string[] = ['hours', 'minutes'],
+  ) => {
+    return to.diff(from, unit).toObject();
+  };
+
+  // @TODO get from utils
+  intervalToString = (
+    interval?: Duration,
+    type?: 'days' | 'hours' = 'hours',
+  ) => {
+    if (!interval) {
+      return '';
+    }
+    if (type === 'days') {
+      switch (interval.days) {
+        case 0:
+          return `${Math.floor(interval.hours)} hours`;
+        case 1:
+          return `${interval.days} day`;
+        default:
+          return `${interval.days} days`;
+      }
+    }
+    return (
+      (interval.hours ? interval.hours + 'h' : '') +
+      (interval.minutes ? ' ' + interval.minutes + 'm' : '')
+    );
+  };
+
+  // @TODO Handle cases when we don't have some data from GraphQL/REST server
+  getSanitizedRouteItem = (route: RouteItem) => {
+    const localArrival = route && DateTime.fromISO(route.arrival?.localTime);
+    const localDeparture =
+      route && DateTime.fromISO(route.departure?.localTime);
+    const utcArrival = route && DateTime.fromISO(route.arrival?.utcTime);
+    const utcDeparture = route && DateTime.fromISO(route.departure?.utcTime);
+    const duration =
+      route &&
+      this.intervalToString(this.getDateTimeInterval(utcArrival, utcDeparture));
+    return {
+      arrival: (route && route.arrival?.city) ?? '',
+      arrivalTime:
+        (localArrival && localArrival.toLocaleString(DateTime.TIME_SIMPLE)) ??
+        '',
+      carrier: { code: (route && route.airline) ?? '', name: '' }, // @TODO: get name of the airline
+      tripDate: (localDeparture && localDeparture.toFormat(dateFormat)) ?? '',
+      departure: (route && route.departure?.city) ?? '',
+      departureTime:
+        (localDeparture &&
+          localDeparture.toLocaleString(DateTime.TIME_SIMPLE)) ??
+        '',
+      duration: duration ?? '',
+      id: (route && route.id) ?? '',
+    };
+  };
+
+  getSectorBorderIndex = () => {
+    const { route, routes } = this.props.data;
+    if (!route) {
+      return null;
+    }
+    return route.findIndex(
+      routeItem => routeItem?.departure?.cityCode === routes?.[1]?.[0],
+    );
+  };
+
+  // @TODO: make ConnectionCard more universal (not only for forth and back routes)
+  getSanitizedRoute = (): Array<TripSectorWithId> => {
+    const { route } = this.props.data;
+    if (route) {
+      return route.map((routeItem: RouteItem) =>
+        this.getSanitizedRouteItem(routeItem),
+      );
+    }
+    return [];
+  };
+
+  getSector = (
+    route: Array<TripSectorWithId>, // rename to sanitizeRoute
+    demandedSector: number,
+    sectorBorderIndex: ?number,
+  ): Array<TripSectorWithId> => {
+    if (
+      demandedSector === null ||
```

Comment:
Could you give names to these conditions for the logic to be easier to follow?

Reply:
OK